### DISCUSSION TITLE
Is the player moving?

### DISCUSSION THREAD

**Discussion by Squander12** [MCPHackers/RetroMCP-Java#110] [Q&A]
Yes, it's me again. But I have a little problem.
I wanted to add the mechanic of destroying shoes when I move.
`@Override
	public void moveEntityWithHeading(float moveStrafing, float moveForward) {
		ItemStack boots = inventory.armorInventory[0];
		if(boots != null && boots.getItem() == Item.windBoots) {
			if(moveStrafing != 0 || moveForward != 0) {
				if(ticksExisted % 80 == 0) {
					boots.damageItem(1, this);
					addChatMessage("Client: " + boots.getItemDamage());
					
					if(boots.stackSize == 0) {
						inventory.armorInventory[0] = null;
					}
				}
			}
		}
		
		double var3 = this.posX;
		double var5 = this.posY;
		double var7 = this.posZ;
		super.moveEntityWithHeading(moveStrafing, moveForward);
		this.addMovementStat(this.posX - var3, this.posY - var5, this.posZ - var7);
	}`
`public void moveEntityWithHeading(float moveStrafing, float moveForward) {
		ItemStack boots = inventory.armorInventory[0];
		if(boots != null && boots.getItem() == Item.windBoots) {
			if(moveStrafing != 0 || moveForward != 0) {
				if(ticksExisted % 80 == 0) {
					boots.damageItem(1, this);
					addChatMessages("Server: " + boots.getItemDamage());
					
					if(boots.stackSize == 0) {
						inventory.armorInventory[0] = null;
					}
				}
			}
		}
		
		double var3 = this.posX;
		double var5 = this.posY;
		double var7 = this.posZ;
		super.moveEntityWithHeading(moveStrafing, moveForward);
		this.func_25045_g(this.posX - var3, this.posY - var5, this.posZ - var7);
	}`
The code comes from the EntityPlayer class, it is the same on the server and client. However, there is a small problem. On the server, moveStrafing and moveForward are always zero. So is there any other way to check if the player is not moving?

**Comment by PhoenixVX** [comment]
Why not simply check if the position has changed from the previous positions fields? `prevPosX`, Y, and Z all exist.
